submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,160 @@
+import "../css/vendor/normalize.css";
+import "../css/styles.css";
+
+const USERS_URL =
+  "https://randomuser.me/api/?inc=gender,phone,picture,dob,location,name&results=25";
+
+const contactsField = document.querySelector(".contacts-field");
+const optionsMenu = document.querySelector(".options");
+const searchInput = document.querySelector(".options__search__inpit");
+let originalContacts;
+let contacts;
+let filter = "";
+
+function getContacts(url) {
+  return fetch(url)
+    .then((res) => res.json())
+    .then(({ results }) => (originalContacts = results))
+    .then((originalContacts) => {
+      copyData();
+      renderContacts(originalContacts);
+    })
+    .catch((error) => {
+      console.error("Ошибка:", error);
+    });
+}
+
+function renderContactTemplate(contact) {
+  const { gender, name, location, dob, phone, picture } = contact;
+  return `<div class="contact__item" data-name="${name.first}">
+            <img class="contact__image" src=${picture.large} alt=${name.first}>
+            <h3 class="contact__title">${name.first} ${name.last}</h3>
+            <p class="contact__phone">${phone}</p>   
+            <div class="contact__additional">
+              <div class="contact__additional__gender">
+                <h4 class="additional__gender__title">sex</h4>
+                <p class="additional__gender_item">${gender}</p>
+              </div>
+              <div class="contact__additional__age">
+                <h4 class="additional__age__title">age</h4>
+                <p class="additional__age_item">${dob.age}</p>
+              </div>
+              <div class="contact__additional__location">
+              <h4 class="additional__location__title">location</h4>   
+              <p class="additional__location_item">${location.country}</p>     
+              </div>
+            </div>
+          </div>`;
+}
+
+optionsMenu.addEventListener("click", onOptionsClick);
+searchInput.addEventListener("change", onChangeSearch);
+
+function onChangeSearch() {
+  contacts = contacts.filter((element) => {
+    return element.name.first
+      .toLowerCase()
+      .includes(searchInput.value.toLowerCase());
+  });
+  renderContacts(contacts);
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => renderContactTemplate(contact))
+    .join("");
+}
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+  switch (selectedOption) {
```

Comment:
It's a huge `switch` statement. You should split it into different parts.

Reply:
I thought on the contrary, it's convenient when there is one switch.

Reply:
Nope.

1) It is hard to read 
2) It contains repeated lines of code
3) This function does "a lot more" than it should do :)

Reply:
try with if...else